iceoryx_posh/include/iceoryx_posh/internal/popo/ports/server_port_user.hpp

```diff
@@ -51,9 +52,9 @@ class ServerPortUser : public BasePort
 
     /// @brief Tries to get the next request from the queue. If there is a new one, the ChunkHeader of the oldest
     /// request in the queue is returned (FiFo queue)
-    /// @return optional that has a new chunk header or no value if there are no new requests in the underlying queue,
+    /// @return cxx::expected that has a new RequestHeader if there are new requests in the underlying queue,
     /// ChunkReceiveResult on error
-    cxx::expected<cxx::optional<const RequestHeader*>, ChunkReceiveResult> getRequest() noexcept;
+    cxx::expected<const RequestHeader*, ChunkReceiveResult> getRequest() noexcept;
```

Comment:
In theory every user has to perform a `nullptr` check before accessing the success expected. Would it be possible to return here a `std::reference_wrapper<const RequestHeader>` to avoid exactly this? Or is it a valid use case that the request header is a nullptr?

Reply:
No, it should never be a `nullptr`. Please se my comment [here](https://github.com/eclipse-iceoryx/iceoryx/pull/1033#discussion_r800678244) and let's discuss it there.